### PR TITLE
Noflo 08

### PR DESCRIPTION
**Pull request by trustmaster**
#50 rebased on latest code

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 language: node_js
 node_js:
+<<<<<<< HEAD
 - '4.6'
+=======
```
Comment (bergie) [inline .travis.yml]: Merge conflict here

File: fbp.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+[
```
Comment (bergie) [inline fbp.json]: Should not be in repo. Add to .gitignore

File: package.json [hunk 3]
```diff
@@ -10,8 +10,8 @@
   "license": "BSD-2-Clause",
   "dependencies": {
     "coffee-script": "^1.12.1",
-    "express": "~4.14.0",
-    "noflo": "0.x >= 0.5",
+    "express": "^4.14.0",
+    "noflo": "git+https://github.com/noflo/noflo.git",
```
Comment (bergie) [inline package.json]: Depending on noflo via git won't work

File: spec/Basic.coffee [hunk 4]
```diff
@@ -24,8 +24,8 @@ describe 'A basic Express server in NoFlo', ->
       net = network
       done()
   after (done) ->
-    net.stop() if net
-    done()
+    net.stop ->
```
Comment (bergie) [inline spec/Basic.coffee]: Should handle the error. So, `net.stop done`

File: spec/Combo.coffee [hunk 5]
```diff
@@ -24,8 +24,8 @@ describe 'A Combo router with multiple filters', ->
       net = network
       done()
   after (done) ->
-    net.stop() if net
-    done()
+    net.stop ->
```
Comment (bergie) [inline spec/Combo.coffee]: Same here

File: spec/Router.coffee [hunk 6]
```diff
@@ -24,8 +24,8 @@ describe 'A static Express Router in NoFlo', ->
       net = network
       done()
   after (done) ->
-    net.stop() if net
-    done()
+    net.stop ->
```
Comment (bergie) [inline spec/Router.coffee]: And here

File: test_components/TestFilter.coffee [hunk 7]
```diff
@@ -2,21 +2,35 @@ noflo = require 'noflo'
...
+        datatype: 'array'
+    outPorts:
+      error: datatype: 'object'
+      filters: datatype: 'array'
+
+  c.forwardBrackets =
+    filters: ['filters']
+
+  c.process (input, output) ->
+    return unless input.has 'pass'
```
Comment (bergie) [inline test_components/TestFilter.coffee]: Should use `hasData`

File: package.json (hunk 3 above)
Comment (trustmaster) [inline package.json]: `"0.x >= 0.5"` works neither for code relying on latest start/stop behavior.

Comment (bergie) [inline package.json]: Yep, I'd consider doing `0.x >= 0.8`

**Comment by trustmaster** [comment]
Updated. Tests should pass when NoFlo 0.8 is released.

**Comment by trustmaster** [comment]
@bergie ready for another review

**Pull request merged**